### PR TITLE
Downgrade CE config verify errors

### PR DESCRIPTION
**Pull request by brianhlin**
Now that we support job transforms, it's feasible that an admin won't
have JOB_ROUTER_ENTRIES or JOB_ROUTER_DEFAULTS defined (the latter
being more difficult since we generate them automatically). For now,
just downgrade this to a warning.

Doc URL doesn't exist yet, I've got a separate branch for it. See https://brianhlin.github.io/htcondor-ce/v5/configuration/writing-job-routes/#setting-job-environments for what I plan for it to look like

### CODE REVIEW

**Comment by matyasselmeci** [comment]
Is there any way to detect if they _are_ using `JOB_ROUTER_ROUTE_*` transforms (or at least have them)? (Are those the 'new' way? I don't like the idea of admins getting warnings forever unless they add some dummy backward compat config or something.)

**Comment by brianhlin** [comment]
Yes but I think the changes are too involved for 5.1.0 so I'll make a ticket for that and target it for 5.1.1 or 5.2.0

File: src/verify_ce_config.py [hunk 1]
```diff
@@ -59,19 +59,22 @@ def main():
...
         except KeyError:
-            error("Missing required %s configuration value" % attr)
+            warn("Missing %s configuration value. " % attr +
+                 "Can be safely ignored if using 'JOB_ROUTER_ROUTE_*' ClassAd transforms.")
 
         # store the ads (iterating through ClassAdStringIterator consumes them)
         try:
             parsed_jr_ads[attr] = list(classad.parseAds(config_val))
         except ValueError:
             # We shouldn't ever get here since classad.parseAds() only raises ValueError when it's given
```
Comment (matyasselmeci) [inline src/verify_ce_config.py:69]: "We shouldn't ever get here" implies at least an error, if not an assertion.

File: src/verify_ce_config.py [hunk 2]
```diff
@@ -59,19 +59,22 @@ def main():
...
             # non-string/non-file output and htcondor.param shouldn't contain such values
-            error("Failed to parse %s configuration value" % attr)
+            warn("Failed to parse %s configuration value. " % attr +
+                 "Can be safely ignored if using 'JOB_ROUTER_ROUTE_*' ClassAd transforms.")
 
         # If JRD or JRE can't be parsed, the job router can't function
         if not parsed_jr_ads[attr]:
-            error("Could not read %s in the HTCondor-CE configuration." % attr)
+            warn("Could not read %s in the HTCondor-CE configuration. " % attr +
```
Comment (matyasselmeci) [inline src/verify_ce_config.py:76]: Is the admin going to get multiple warnings for the same thing?

File: src/verify_ce_config.py (hunk 1 above)
Comment (brianhlin) [reply src/verify_ce_config.py:69]: Agreed but I'm not keen on keeping this an error that could prevent CE startup without knowing more about how an unset `JOB_ROUTER_ENTRIES` interacts with this piece of code.

File: src/verify_ce_config.py (hunk 2 above)
Comment (brianhlin) [reply src/verify_ce_config.py:76]: Yes, once each for an unparseable JRE and JRD. The admin will really have to go out of their way to unset the latter, though.

File: src/verify_ce_config.py (hunk 1 above)
Comment (matyasselmeci) [reply src/verify_ce_config.py:69]: Can you update the comment then?

Comment (matyasselmeci) [reply src/verify_ce_config.py:69]: Actually, now that KeyError doesn't exit (?), is it still true that we shouldn't get to the ValueError exception?
Or, worse, can
```
            parsed_jr_ads[attr] = list(classad.parseAds(config_val))
```
crash?

Comment (brianhlin) [reply src/verify_ce_config.py:69]: D'oh, good point

**Comment by matyasselmeci** [comment]
It kinda feels like we're warning people for doing the right thing. We could also turn the warnings into debug messages until we get JOB_ROUTER_ROUTE detection done -- I'll encourage us to hurry with it :-)

The code looks overall fine except you should make sure that changes to the code flow caused by not bailing early won't cause exceptions at later points.

**Comment by brianhlin** [comment]
@matyasselmeci updated the commit to address your concerns

**Review by matyasselmeci — APPROVED** [review]

**Pull request merged**